### PR TITLE
[webui] add basename in router

### PR DESCRIPTION
**Pull request by Lijiaoa**
related issue: #2771 

With this PR merged, you could run an experiment with command `nnictl create --config xx.yml --url_prefix customURL` open webui website look like this : `ip:port/customURL/oview`

- [x] test with **Hao Ni**'s PR #3643
- [x] Rest api: `/api/v1/nni${prefix}`, 如果用户未设置prefix，`api/v1/nni`
- [x] `/logs api` needs prefix? -> no need to add prefix for /logs api

### 测试用例
（1）`nnictl create --config xx.yml` 起一个实验，观察 ip:port/oview、 /detail、 /experiment 页面是否正常
（2）`nnictl create --config xx.yml --url_prefix customPrefix` 起一个实验，观察 ip:port/customPrefix/oview、/detail、 /experiment 页面是否正常
（3）`nnictl create --config xx.yml --url_prefix /customerURL` 起一个实验，此时会提醒不能以 `/`  开头，路径无效
（4）`nnictl create --config xx.yml --url_prefix customer/xxx` 起一个实验，观察 ip:port/customPrefix/oview、/detail、 /experiment 页面是否正常

### CODE REVIEW

File: ts/webui/src/static/function.ts [hunk 1]
```diff
@@ -4,6 +4,17 @@ import { IContextualMenuProps } from '@fluentui/react';
 import { MANAGER_IP } from './const';
 import { MetricDataRecord, FinalType, TableObj, Tensorboard } from './interface';
 
+function getPrefix(): string | undefined {
+    const pathName = window.location.pathname;
+    let newPathName = pathName;
+
+    if (pathName.endsWith('/oview' || '/detail' || '/experiment')) {
```
Comment (ultmaster) [inline ts/webui/src/static/function.ts:11]: Shouldn't it be:

`pathName.endsWith('/oview') || pathName.endsWith('/detail') || pathName.endsWith('/experiment')`

File: ts/webui/src/components/stateless-component/NNItabs.tsx [hunk 2]
```diff
@@ -2,19 +2,19 @@ import * as React from 'react';
 import { NavLink } from 'react-router-dom';
 
 const OVERVIEWTABS = (
-    <NavLink to={'/oview'} activeClassName='selected' className='common-tabs'>
+    <NavLink to='/oview' activeClassName='selected' className='common-tabs'>
```
Comment (ultmaster) [inline ts/webui/src/components/stateless-component/NNItabs.tsx:5]: What's the difference here?

File: ts/webui/src/static/function.ts [hunk 3]
```diff
@@ -4,6 +4,17 @@ import { IContextualMenuProps } from '@fluentui/react';
...
+        newPathName = pathName.replace('/oview' || '/detail' || '/experiment', '');
```
Comment (ultmaster) [inline ts/webui/src/static/function.ts:12]: What if the prefix is `/experiment-number-1`? Will it be mistakenly replaced?

Comment (Lijiaoa) [reply ts/webui/src/static/function.ts:12]: it doesn't support prefix with '/' on the head.
image

File: ts/webui/src/components/stateless-component/NNItabs.tsx (hunk 2 above)
Comment (Lijiaoa) [reply ts/webui/src/components/stateless-component/NNItabs.tsx:5]: {} 配合变量使用，这里直接写字符串就行

File: ts/webui/src/static/function.ts (hunk 1 above)
Comment (Lijiaoa) [reply ts/webui/src/static/function.ts:11]: will change it at once! it's my oversight. Thanks!

**Review by acured — APPROVED** [review]

File: ts/webui/src/static/const.ts [hunk 4]
```diff
@@ -1,10 +1,17 @@
+import { getPrefix } from './function';
+
 // when there are more trials than this threshold, metrics will be updated in group of this size to avoid freezing
 const METRIC_GROUP_UPDATE_THRESHOLD = 100;
 const METRIC_GROUP_UPDATE_SIZE = 20;
 
-const MANAGER_IP = `/api/v1/nni`;
+const prefix = getPrefix();
+
+const MANAGER_IP = prefix === undefined ? '/api/v1/nni' : `/api/v1/nni${prefix}`;
```
[the remark this answers is not in the document]
Comment (Lijiaoa) [reply ts/webui/src/static/const.ts]: ok

File: ts/webui/src/static/function.ts (hunk 1 above)
Comment (Lijiaoa) [reply ts/webui/src/static/function.ts]: had been fixed

**Review by ultmaster — APPROVED** [review]

**Review by QuanluZhang — APPROVED** [review]

**Pull request merged**